pkg/new-ui/v1beta1/backend.go

```diff
@@ -302,7 +456,14 @@ func (k *KatibUIHandler) DeleteTemplate(w http.ResponseWriter, r *http.Request)
 	updatedConfigMapName := data["updatedConfigMapName"].(string)
 	updatedConfigMapPath := data["updatedConfigMapPath"].(string)
 
-	newTemplates, err := k.updateTrialTemplates(updatedConfigMapNamespace, updatedConfigMapName, "", updatedConfigMapPath, "", ActionTypeDelete)
+	err = IsAuthorized(user, "delete", updatedConfigMapNamespace, "", "v1", "configmaps", "", updatedConfigMapName, &k.sarClient)
+	if err != nil {
+		log.Printf("The user: %s is not authorized to delete configmap: %s from namespace: %s \n", user, updatedConfigMapName, updatedConfigMapNamespace)
+		http.Error(w, err.Error(), http.StatusForbidden)
+		return
+	}
+
+	newTemplates, err := k.updateTrialTemplates(updatedConfigMapNamespace, updatedConfigMapName, "", updatedConfigMapPath, "", ActionTypeDelete, user)
```

Comment:
@kimwnasptd The updateTrialTemplate() function will first try to retrieve the TrialTemplate ConfigMap and then perform update/delete/create depending on the route (Edit/Add/DeleteTemplate) from which it is called. 

The current implementation will either delete the TrialTemplate ConfigMap if its empty or update the ConfigMap.
Currently we check only if a user has permissions to delete ConfigMaps. Should we move the authorization checks  inside updateTrialTemplate()?